### PR TITLE
Use new cache endpoint for prebid mobile

### PR DESCRIPTION
**Pull request by nhedley**
Use `/cache` instead of `/put` for Prebid Server to cache bids on Prebid Cache for mobile requests. The new endpoint requires `type: "json"` and the value is json instead of a string.

### CODE REVIEW

File: pbs_light.go [hunk 1]
```diff
@@ -362,18 +354,14 @@ func auction(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
...
-			bc := BidCache{
+			bc := &pbc.BidCache{
 				Adm:    bid.Adm,
 				NURL:   bid.NURL,
 				Width:  bid.Width,
 				Height: bid.Height,
 			}
-			buf := new(bytes.Buffer)
-			enc := json.NewEncoder(buf)
-			enc.SetEscapeHTML(false)
```
Comment (dbemiller) [inline pbs_light.go]: This is a small/subtle difference from the current behavior, and I don't see a test for it... so just making sure it's ok.

`prebid_cache.go` transforms these objects into JSON with: `reqJSON, err := json.Marshal(pr)`

The impementation of json.Marshal is:

```
	e := &encodeState{}
	err := e.marshal(v, encOpts{escapeHTML: true})
```

The code you're deleting here has `enc.SetEscapeHTML(false)`.

Does this sound like trouble to you, or are you sure it's safe? I'm suspecting that most ad markup (`adm`) has some HTML in it.

Comment (nhedley) [reply pbs_light.go]: Thanks for catching this, I do think that we need SetEscapeHTML(false). I'll work on the implementation and add tests.

**Comment by mkendall07** [comment]
@nhedley 
Is the app making get requests `/cache` as well? Currently the `/cache` and the `/get/put` endpoints are not compatible.

**Comment by nhedley** [comment]
@mkendall07 The iOS & Android SDKs currently don't call Prebid Cache. Only pbm.js calls `/get` right now. I've made the updates for pbm.js to use `/cache` instead of `/get` and spoke with @dbemiller about making sure to do pbm.js release in conjunction with Prebid Server release so we shouldn't have any issues.

**Pull request merged**